pkg/controller/plan/kubevirt.go

```diff
@@ -1279,6 +1284,67 @@ func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualM
 	return
 }
 
+func (r *KubeVirt) setIntanceType(vm *plan.VMStatus, object *cnv.VirtualMachine) (virtualMachine *cnv.VirtualMachine, err error) {
+	if r.Plan.Spec.SelectedInstanceType == "" {
+		return object, nil
+	}
+	intancetypeName, kind, err := r.getInstancetype(vm, r.Plan.Spec.SelectedInstanceType)
+	if err != nil {
+		return
+	}
+	object.Spec.Instancetype = &cnv.InstancetypeMatcher{Name: intancetypeName, Kind: kind}
+	virtualMachine = object
+	return
+}
+
+// Attempt to find a suitable instance type
+func (r *KubeVirt) getInstancetype(vm *plan.VMStatus, instancetypeName string) (name, kind string, err error) {
+	name, kind, err = r.getVirtualMachineInstancetype(instancetypeName)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			r.Log.Info("could not find a local instance type for destination VM. trying cluster wide",
```

Comment:
this logic is a bit weird to me - first, since it assumes the instance type resides in the target namespace and second, because we can't distinguish between namespaced and cluster instancetype with the same name
we should probably add also a namespace to the instance type on the plan

Reply:
it's clear that the namespace will be the target namespace. we can't use anything else if the vm is planned to exist there

Reply:
I'm not sure - did you try to use an instance type from a different namespace that you have permissions on?

Reply:
@0xFelix, I see that indeed [InstacetypeMatcher has no namespace](https://github.com/kubevirt/kubevirt/blob/74f5cb9bcfee98661548ae5dfe900430d8cee9f9/staging/src/kubevirt.io/api/core/v1/types.go#L2835-L2868), so users can use instacetypes they have on other namespaces they have permissions on?

Reply:
@ahadas Right now it is possible to use either use cluster wide instancetypes or instancetypes in the same namespace as the VM. Cross-namespace access to instancetypes is not possible.

Reply:
ack
@liranr23 we should keep this in mind when we get to document it then because I suppose ordinary users won't be provided with permissions to create cluster-level instance type so in order to use a custom instance type, users would have to create a namespace and write or copy an instance type such that it would exist in the target namespace..